### DISCUSSION TITLE
Unrecognized circle appears

### DISCUSSION THREAD

**Discussion by bimintelligence** [IfcOpenShell/IfcOpenShell#2687] [Q&A]
Hi, an IFC was established for hollow cylindrical surface, but there is unexpected circle in the IFCVIEWER.
![image](https://user-images.githubusercontent.com/73410751/211745239-7f0cdd45-a6aa-40fa-a69d-68b5937201f9.png)

[hollowCylindricalSurface.ifc.txt](https://github.com/IfcOpenShell/IfcOpenShell/files/10389770/hollowCylindricalSurface.ifc.txt)

**Comment by Moult** [comment]
Can you share a valid file? The file you have is missing the header which makes it unparsable, which suggests that other portions of the file might also be missing.

**Reply by bimintelligence** [reply]
Thanks for your attention. Here's the outputted IFC file with the header.
The IFC file is initialized by my self, and I parsed in IfcViewer of rdf co.
![image](https://user-images.githubusercontent.com/73410751/211990570-43cba104-0344-4995-8eea-f4ac0de6d2ca.png)

[hollowCylindricalSurface.ifc.txt](https://github.com/IfcOpenShell/IfcOpenShell/files/10398671/hollowCylindricalSurface.ifc.txt)

**Reply by aothms** [reply]
I think you need to discuss this with @peterrdf. If I look at the placements of circles and ellipses, they seem to be on a single orthogonal line.

```
>>> import numpy
>>> import ifcopenshell
>>> f = ifcopenshell.open('hollowCylindricalSurface.ifc')
>>> numpy.array([x.Position.Location[0] for x in f.by_type('IfcCircle') + f.by_type('IfcEllipse')])
array([[-1.16000e+08, -1.25603e+08,  5.48000e+08],
       [-1.16000e+08, -3.26030e+07,  5.48000e+08],
       [-1.16000e+08, -3.26030e+07,  5.48000e+08],
       [-1.16000e+08, -1.25603e+08,  5.48000e+08],
       [-1.16000e+08, -3.26030e+07,  5.48000e+08],
       [-1.16000e+08, -3.26030e+07,  5.48000e+08],
       [-1.16000e+08, -1.25603e+08,  5.48000e+08],
       [-1.16000e+08, -1.25603e+08,  5.48000e+08]])
```

**Comment by peterrdf** [comment]
There are a few issues with the new example file:
- the top or bottom has a misplacement (just one of the 4 brep faces)
- the cylindrical surfaces are incorrectly cutted through the polygon

You can look in detail for issues using our 3DEditor for generated .bin / .rdf files: https://rdf.bg/download/3deditor-sources-20221121.zip

Such .rdf / .bin files can be created directly selecting the object in the 3D pane of our IFC Viewer (https://rdf.bg/product-list/ifc-engine/ifc-viewer/). It is also possible to generate such .rdf / .ifc files from any instance in the IFC file if it contains geometry (for example just focussing on one ifc advanced face), for this the IFCEditor can be used. 
<img width="1384" alt="advancedfacebound" src="https://user-images.githubusercontent.com/6512688/212343862-dad3fd86-92c6-4bda-a22a-4ce9726672f0.png">
